src/main/java/io/cdap/cdap/events/process/pubsub/PubSubEventWriter.java

```diff
@@ -0,0 +1,206 @@
+            // This means to configure the proxy if it comes from the CDAP configuration
+            if (proxyHost != null && proxyPort != null) {
+                configureChannelProxy(publisherBuilder, proxyHost, proxyPort);
+            }
+            this.publisher = publisherBuilder.build();
+            logger.info("Publisher created successfully");
+        } catch (IOException e) {
+            logger.error("Error creating pubsub events.publisher. Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Method to use the service account path as a File
+     *
+     * @param saPath Service account where is storage
+     * @return Service account as a file
+     * @throws FileNotFoundException If the file does not exists, it will throws a FileNotFoundException
+     */
+    private InputStream getCredentials(String saPath) throws FileNotFoundException {
+        File credentialsFile = new File(saPath);
+        return new FileInputStream(credentialsFile);
+    }
+
+    /**
+     * Method to configure proxy to the Pub/Sub Builder
+     *
+     * @param builder   Pub/Sub builder where was already initialized with the topic name, and project
+     * @param proxyHost Proxy host where configure the use of the proxy
+     * @param proxyPort Proxy port where configure the use of the proxy
+     * @throws NumberFormatException
+     */
+    private void configureChannelProxy(Publisher.Builder builder, String proxyHost, String proxyPort)
+            throws NumberFormatException {
+        SocketAddress proxySocketAddress = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
+        builder.setChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                        .setChannelConfigurator(managedChannelBuilder -> managedChannelBuilder.proxyDetector(
+                                new ProxyDetector() {
+                                    @Nullable
+                                    @Override
+                                    public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
+                                        if (socketAddress == null) {
+                                            return null;
+                                        }
+                                        return HttpConnectProxiedSocketAddress.newBuilder()
+                                                .setTargetAddress((InetSocketAddress) socketAddress)
+                                                .setProxyAddress(proxySocketAddress)
+                                                .build();
+                                    }
+                                }))
+                        .build()
+        );
+    }
+
+    @Override
+    public void publishEvent(Event event) {
+        if (publisher == null) {
+            logger.info("Publisher is not already initialized");
+            return;
+        }
+        logger.info("Publishing event");
```

Comment:
Publish will be a very frequent operation. This message could flood the log. Remove it or make it trace.